File: sdks/php/src/Api/OperationalWebhook.php
<?php
declare(strict_types=1);
namespace HookSniff\Api;
use HookSniff\Request\HookSniffHttpClient;

class OperationalWebhook {
    private HookSniffHttpClient $client;
    public function __construct(HookSniffHttpClient $client) { $this->client = $client; }
    public function list(): array { return $this->client->request('GET', '/api/v1/operational-webhooks'); }
    public function create(array $body): array { return $this->client->request('POST', '/api/v1/operational-webhooks', $body); }
    public function get(string $id): array { return $this->client->request('GET', "/api/v1/operational-webhooks/{$id}"); }
    public function update(string $id, array $body): array { return $this->client->request('PUT', "/api/v1/operational-webhooks/{$id}", $body); }
    public function delete(string $id): void { $this->client->request('DELETE', "/api/v1/operational-webhooks/{$id}"); }
    public function listDeliveries(string $id): array { return $this->client->request('GET', "/api/v1/operational-webhooks/{$id}/deliveries"); }
}

File: sdks/rust/src/api/operational_webhook.rs
use crate::api_internal::HookSniffHttpClient;
use crate::error::Error;
use crate::models::{OperationalWebhookDeliveryOut, OperationalWebhookEndpointIn, OperationalWebhookEndpointOut};

pub struct OperationalWebhook<'a> { client: &'a HookSniffHttpClient }

impl<'a> OperationalWebhook<'a> {
    pub fn new(client: &'a HookSniffHttpClient) -> Self { Self { client } }

    pub async fn list(&self) -> Result<Vec<OperationalWebhookEndpointOut>, Error> {
        self.client.get("/api/v1/operational-webhooks").await
    }
    pub async fn create(&self, body: &OperationalWebhookEndpointIn) -> Result<OperationalWebhookEndpointOut, Error> {
        self.client.post("/api/v1/operational-webhooks", body).await
    }
    pub async fn get(&self, id: &str) -> Result<OperationalWebhookEndpointOut, Error> {
        self.client.get(&format!("/api/v1/operational-webhooks/{}", id)).await
    }
    pub async fn update(&self, id: &str, body: &OperationalWebhookEndpointIn) -> Result<OperationalWebhookEndpointOut, Error> {
        self.client.put(&format!("/api/v1/operational-webhooks/{}", id), body).await
    }
    pub async fn delete(&self, id: &str) -> Result<(), Error> {
        self.client.delete(&format!("/api/v1/operational-webhooks/{}", id)).await
    }
    pub async fn list_deliveries(&self, id: &str) -> Result<Vec<OperationalWebhookDeliveryOut>, Error> {
        self.client.get(&format!("/api/v1/operational-webhooks/{}/deliveries", id)).await
    }
}
